pygmt/helpers/decorators.py

```diff
@@ -760,14 +766,23 @@ def new_module(*args, **kwargs):
                 if fmt in separators and is_nonstr_iter(value):
                     for index, item in enumerate(value):
                         if " " in str(item):
-                            # Check if there is a space " " when converting
-                            # a pandas.Timestamp/xr.DataArray to a string.
-                            # If so, use np.datetime_as_string instead.
-                            # Convert datetime-like item to ISO 8601
-                            # string format like YYYY-MM-DDThh:mm:ss.ffffff.
-                            value[index] = np.datetime_as_string(
-                                np.asarray(item, dtype=np.datetime64)
-                            )
+                            # Check if there is a space " " in the item, which
+                            # is typically present in objects such as
+                            # np.timedelta64, pd.Timestamp, or xr.DataArray.
+                            # If so, convert the item to a numerical or string
+                            # type that is understood by GMT as follows:
+                            if getattr(
+                                getattr(item, "dtype", ""), "name", ""
+                            ).startswith("timedelta"):
+                                # A np.timedelta64 item is cast to integer
+                                value[index] = item.astype("int")
+                            else:
+                                # A pandas.Timestamp/xr.DataArray containing
+                                # a datetime-like object is cast to ISO 8601
+                                # string format like YYYY-MM-DDThh:mm:ss.ffffff
+                                value[index] = np.datetime_as_string(
+                                    np.asarray(item, dtype=np.datetime64)
+                                )
```

Comment:
The if-then logic here might get a lot more complicated if we also decide to support passing PyArrow time/date types (xref #2800) into `region`/`-R`. At such, it might be good to isolate this logic into a separate function instead, so I'm thinking of maybe reverting commit d5bedc2ac71a337f2b2952843778d2e8c78c29a5 for now, but cherry-pick that change into a separate PR where we create a dedicated `_cast_datetime_to_str` function.

Reply:
Sounds good.

Reply:
Ok, commit reverted in 849feda91c37a4ca2b449309b518c0a9310e5820. Should be ready to review now.